packages/api-core/src/api.js

```diff
@@ -153,32 +152,23 @@ export default class AvApi {
             resolve,
             newConfig.pollingIntervals[newConfig.attempt] || 1000
           );
-        }).then(() => this.request(newConfig, afterResponse));
+        }).then(() => this.request(newConfig));
       }
     }
 
-    return afterResponse ? afterResponse(response) : response;
-  }
-
-  getError(error) {
-    const response = {};
-    response.original = error;
-    response.code = get(error, 'response.status');
-    response.message = get(error, 'response.statusText');
-    response.url = get(error, 'config.url');
     return response;
   }
 
   // make request to http
-  request(config, afterResponse) {
+  request(config) {
     if (config.polling) {
       config.attempt = config.attempt || 0;
       config.attempt += 1;
     }
 
     return this.http(config)
-      .then(response => this.onResponse(response, afterResponse))
-      .catch(error => this.Promise.reject(this.getError(error)));
+      .then(response => this.onResponse(response))
+      .catch(error => this.Promise.reject(error));
```

Comment:
just don't catch the error and the error will bubble up since the promise is already being returned.

Reply:
agreed. awesome feedback. 🙏